services/supervisor/src/simple/schema.rs

```diff
@@ -14,25 +14,37 @@
 
 use exonum::{
     crypto::Hash,
-    merkledb::{Entry, IndexAccess, ObjectHash},
+    merkledb::{AccessExt, Entry, IndexAccessMut, ObjectHash},
 };
 
 use super::ConfigPropose;
 
-pub struct Schema<T: IndexAccess> {
-    access: T,
+const NOT_INITIALIZED: &str = "Supervisor schema is not initialized";
+
+pub struct Schema<T: AccessExt> {
+    pub config_propose: Entry<T::Base, ConfigPropose>,
 }
 
-impl<T: IndexAccess> Schema<T> {
+impl<T: AccessExt> Schema<T> {
```

Comment:
It could make sense to restrict service schema constructors to `Prefixed<_>` in order to minimize possible misuse, like `SomeSchema::new(&testkit.snapshot())`.